src/train_cmd.cpp

```diff
@@ -3866,6 +3868,9 @@ static TileIndex TrainApproachingCrossingTile(const Train *v)
  */
 static bool TrainCheckIfLineEnds(Train *v, bool reverse)
 {
+
+	if (v->flags.Test(VehicleRailFlag::Reversing)) return false;
```

Comment:
Extra newline, could also use a comment explaining why we're checking this.

```suggestion
	if (v->flags.Test(VehicleRailFlag::Reversing)) return false;
```